### PR TITLE
feat(feature-toggle): added a handler service that calls the specific handler related to feature

### LINKED ISSUE #1447: A feature handler to manage specific config or settings at feature level
**Is your feature request related to a problem? Please describe.**
Right now in the feature toggle we just check if the feature is allowed or not, we dont perform any handling on the feature.

**Describe the solution you'd like**
Include a handler that can be invoked in case any additional handling is needed after checking the feature allowance.

### PR DESCRIPTION
**Pull request by yeshamavani**
MIGRATION CHANGE:
migration-20230612141912- feature level metadata

gh-1447

## Description

After validating the feature checking if any handler is required for the particular feature. 
For handling the allowed feature an extension point is used that requires the handler name as argument to call the specific handler. There is a handler interface each and every handler should implement. All the custom handler logic can be added in the handler.

Fixes #1447 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/cli/README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ $ npm install -g @sourceloop/cli
 $ sl COMMAND
 running command...
 $ sl (-v|--version|version)
-@sourceloop/cli/4.0.4 darwin-arm64 node-v18.16.0
+@sourceloop/cli/4.0.4 darwin-arm64 node-v16.20.0
```
Comment (samarpan-b) [inline packages/cli/README.md:21]: why this change ?

File: packages/feature-toggle/src/providers/feature-flag-action.provider.ts [hunk 2]
```diff
@@ -32,15 +35,28 @@ export class FeatureFlagActionProvider implements Provider<FeatureFlagFn> {
...
+      const isAllowed =
+        intersection(featureToCheck, disabledFeatures).length <= 0;
+      if (!isAllowed) {
+        return false;
+      } else {
+        // check for handler
+        const handler = metadata.options?.handler;
+
+        if (handler) {
+          await this.handlerService?.handle(handler);
```
Comment (samarpan-b) [inline packages/feature-toggle/src/providers/feature-flag-action.provider.ts:54]: This handler should also return a result which could be boolean

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: packages/feature-toggle/src/providers/feature-flag-action.provider.ts (hunk 2 above)
Comment (yeshamavani) [reply packages/feature-toggle/src/providers/feature-flag-action.provider.ts]: done
